packages/core/src/template-compiler.ts

```diff
@@ -316,12 +317,19 @@ export default class TemplateCompiler {
     }
 
     let opts = this.syntax.defaultOptions({ contents, moduleName });
-    opts.plugins!.ast = [...this.params.plugins.ast!.slice().reverse(), ...opts.plugins!.ast!];
+    if (this.params.resolver) {
+      let transform = this.params.resolver.astTransformer(this);
+      if (transform) {
+        this.params.plugins.ast!.push(transform);
+      }
+    }
+    opts.plugins!.ast = [...this.getReversedASTPlugins(this.params.plugins.ast!), ...opts.plugins!.ast!];
```

Comment:
A brief comment describing the exact precedence/ordering here would be good. It is a little tricky, so us sharing that knowledge will help future travelers.